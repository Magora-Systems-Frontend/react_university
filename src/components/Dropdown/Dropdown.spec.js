import React from 'react';
import { shallow, configure } from 'enzyme';
import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import Component from './index';

configure({ adapter: new Adapter() });

describe('component snapshot', () => {
  it('capturing snapshot of dropdown', () => {
    const renderedValue = renderer.create(<Component displayBubble={true} />).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
});

describe('Shallow render component', () => {
  let wrapper;

  const initialState = {
    displayBubble: false,
  };

  beforeEach(() => {
    wrapper = shallow(<Component />);
  });

  it('check initial state', () => {
    expect(wrapper.state()).toEqual(initialState);
  });

  it('check change state', () => {
    wrapper.setState({ displayBubble: true });
    expect(wrapper.state()).toEqual({ displayBubble: true });
  });

  it('check toggle functions', () => {
    wrapper.instance().toggleBubble();
    expect(wrapper.state()).toEqual({ displayBubble: true });
    wrapper.instance().toggleBubble();
    expect(wrapper.state()).toEqual({ displayBubble: false });
  });

  it('check events handle', () => {
    wrapper.simulate('mouseenter');
    expect(wrapper.state()).toEqual({ displayBubble: true });
    wrapper.simulate('mouseleave');
    expect(wrapper.state()).toEqual({ displayBubble: false });
  });

  it('check bubble position', () => {
    wrapper.setProps({ bubblePosition: 'left' });
    wrapper.setState({ displayBubble: true });
    expect(wrapper.find('.dropdown__bubble').hasClass('dropdown__bubble_right')).toBe(true);
  });

  it('check mobile class', () => {
    wrapper.setProps({ isMobile: true });
    wrapper.setState({ displayBubble: true });
    expect(wrapper.find('.dropdown__bubble').hasClass('dropdown__bubble_mobile')).toBe(true);
  });
});
