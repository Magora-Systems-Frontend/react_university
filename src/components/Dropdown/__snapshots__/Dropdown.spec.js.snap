// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component snapshot capturing snapshot of dropdown 1`] = `
<div
  className="dropdown"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <div
    className="dropdown__bubble false undefined"
    style={
      Object {
        "width": undefined,
      }
    }
  />
</div>
`;
